src/swupd_lib/target_root.c

```diff
@@ -732,21 +784,27 @@ enum swupd_code target_root_install_files(struct list *files, struct manifest *m
 int target_root_remove_files(struct list *files)
 {
 	struct list *iter = NULL;
+	struct list *dirs_to_remove = NULL;
+	struct list *dirs_iter = NULL;
 	struct file *file = NULL;
 	char *fullfile = NULL;
 	int total = list_len(files);
 	int deleted = total;
 	int count = 0;
+	int ret;
 
 	iter = list_head(files);
 	while (iter) {
 		file = iter->data;
 		iter = iter->next;
 		fullfile = sys_path_join("%s/%s", globals.path_prefix, file->filename);
-		if (sys_rm_recursive(fullfile) == -1) {
-			/* if a -1 is returned it means there was an issue deleting the
-			 * file or directory, in that case decrease the counter of deleted
-			 * files.
+		ret = sys_rm(fullfile);
+		if (ret == -ENOTEMPTY) {
+			count--;
```

Comment:
I don't understand why we're decrementing `count` here. Maybe add comments to explain the use and relationship of `count`, `total`, and `deleted`?

Reply:
So I think what's going on here is that count is how many files the loop has processed so far and I basically am decrementing it because in this case we still need to process it in the next loop (there is an unconditional increment below).

deleted starts as total and is decremented each time a file fails to be deleted.

I can definitely add a bit of documentation around that, this is all for book keeping reporting to the user (deleted should match total for a successful operation).